src/main.ts

```diff
@@ -10,44 +10,50 @@ import post from './post';
 import { PullRequest } from './pull-request';
 import { isPost } from './state';
 
-let pr: PullRequest | undefined = undefined;
-
-// All the code should be inside this try block
-try {
-  const octokit = getOctokit(getInput('github_token', { required: true }));
-
-  if (!context.issue || !context.issue.number) {
-    warning('Pull request statuses are not available in this context');
-    info('No issue number found in the context');
-
-    // Create "empty" PullRequest object
-    pr = new PullRequest(undefined, undefined, octokit);
-  } else {
-    pr = await PullRequest.initialize(context.issue.number, octokit);
-  }
-
-  // Check if the script was invoked in the post step
-  if (!isPost) {
-    // Call the action function from action.ts
-    await action(pr);
-  } else {
-    await post(pr);
+async function run(): Promise<void> {
+  let pr: PullRequest | undefined = undefined;
+
+  // All the code should be inside this try block
+  try {
+    const octokit = getOctokit(getInput('github_token', { required: true }));
+
+    if (!context.issue || !context.issue.number) {
+      warning('Pull request statuses are not available in this context');
+      info('No issue number found in the context');
+
+      // Create "empty" PullRequest object
+      pr = new PullRequest(undefined, undefined, octokit);
+    } else {
+      pr = await PullRequest.initialize(context.issue.number, octokit);
+    }
+
+    // Check if the script was invoked in the post step
+    if (!isPost) {
+      // Call the action function from action.ts
+      await action(pr);
+    } else {
+      await post(pr, octokit);
+    }
+  } catch (error) {
+    let message: string;
+
+    if (error instanceof Error) {
+      message = error.message;
+    } else {
+      message = JSON.stringify(error);
+    }
+
+    // Set the Pull Request status to error when error occurs
+    if (pr && pr.isInitialized()) {
+      const url = error instanceof TFError ? error.url : undefined;
+      await pr.setStatus('error', `${message}`, url);
+    }
+
+    // Log the error and set the action status to failed
+    setFailed(message);
   }
-} catch (error) {
-  let message: string;
+}
 
-  if (error instanceof Error) {
-    message = error.message;
-  } else {
-    message = JSON.stringify(error);
-  }
+await run();
```

Comment:
I don't follow how this helps to resolve issue.

Reply:
Would it be possible add before `await run();` some comment for user who are not familiar with the GitHub Actions?

Reply:
Why is it present as a function.